core/src/impl/Kokkos_Stacktrace.cpp

```diff
@@ -0,0 +1,248 @@
+#include "Kokkos_Macros.hpp"
+#include "Kokkos_Stacktrace.hpp"
+
+#ifdef KOKKOS_IMPL_ENABLE_STACKTRACE
+// backtrace() function for retrieving the stacktrace
+#include <execinfo.h>
+#endif
+#ifdef KOKKOS_IMPL_ENABLE_CXXABI
+#include <cxxabi.h>
+#endif  // KOKKOS_ENABLE_CXXABI
+
+#include <exception>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+namespace Kokkos {
+namespace Impl {
+#ifndef KOKKOS_IMPL_ENABLE_STACKTRACE
+int backtrace(void**, int) { return 0; }
+char** backtrace_symbols(void* const*, int) { return NULL; }
+#endif
+
+std::string demangle(const std::string& name) {
+#ifndef KOKKOS_IMPL_ENABLE_CXXABI
+  return name;
+#else
+  size_t found_end = name.find_first_of("+)", 0, 2);
+  if (found_end == std::string::npos) {
+    found_end = name.size();
+  }
+  size_t found_parenthesis = name.find_first_of("(");
+  size_t start             = found_parenthesis + 1;
+  if (found_parenthesis == std::string::npos) start = 0;
+
+  std::string s = name.substr(start, found_end - start);
+
+  if (s.length() != 0) {
+    int status          = 0;
+    char* output_buffer = nullptr;
+    size_t length       = s.length();
+    char* d = abi::__cxa_demangle(s.c_str(), output_buffer, &length, &status);
+    if (d != nullptr) {
+      s = d;
+      free(d);
+    }
+  }
+
+  // Special cases for "main" and "start" on Mac
+  if (s.length() == 0) {
+    if (name == "main" || name == "start") {
+      s = name;
+    }
+  }
+  return s;
+#endif  // KOKKOS_ENABLE_CXXABI
+}
+
+class Stacktrace {
+ public:
+  Stacktrace()                  = delete;
+  Stacktrace(const Stacktrace&) = delete;
+  Stacktrace& operator=(const Stacktrace&) = delete;
+  Stacktrace(Stacktrace&&)                 = delete;
+  Stacktrace& operator=(Stacktrace&&) = delete;
+  ~Stacktrace()                       = delete;
+
+  // These are public only to avoid wasting an extra stacktrace line.
+  // See save_stacktrace below.
+  static constexpr int capacity = 100;
```

Comment:
Why isn't this configurable?

Reply:
I think its fine to just hardcode it for now. I don't want another CMake option if we can avoid it, and 100 deep stack is probably fine ...